Mingw out of source build

**Please acknowledge the following before creating a ticket**

- [X ] I have read the GitHub issues section of [REPORTING-BUGS](../blob/master/REPORTING-BUGS).

**Description of the bug:**
When building out of source with mingw, the relative include path for `os/posix/include` is not found and the build fails to find things like `sys/socket.h`

**Environment**: CentOS Stream 9 + mingw64 gcc version 14.2.1

**fio version**: Reproduced on 3.35 and 3.39

**Reproduction steps**
```
[dev@d-lc9-kai fio]$ git status
HEAD detached at fio-3.39
nothing to commit, working tree clean
[dev@d-lc9-kai fio]$ mkdir build
[dev@d-lc9-kai fio]$ cd build/
[dev@d-lc9-kai build]$ CROSS_COMPILE=x86_64-w64-mingw32- ../configure
....
[dev@d-lc9-kai build]$ make V=1
....
x86_64-w64-mingw32-gcc -o crc/crc32c-arm64.o -DFIO_VERSION='"fio-3.39"' -std=gnu99 -Wwrite-strings -Wall -Wdeclaration-after-statement -g -ffast-math  -D_GNU_SOURCE -include config-host.h  -D_WIN32_WINNT=0x0601 -ffunction-sections -fdata-sections -Wimplicit-fallthrough -I. -I.. -O3 -march=native -DPSAPI_VERSION=1 -Ios/windows/posix/include -Wno-format -D_GNU_SOURCE -include config-host.h  -D_WIN32_WINNT=0x0601 -ffunction-sections -fdata-sections -Wimplicit-fallthrough -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DFIO_INTERNAL -DFIO_INC_DEBUG -DBITS_PER_LONG=32 -c ../crc/crc32c-arm64.c
In file included from ../crc/crc32c-arm64.c:2:
../crc/../os/os.h:5:10: fatal error: sys/socket.h: No such file or directory
    5 | #include <sys/socket.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
make: *** [../Makefile:503: crc/crc32c-arm64.o] Error 1
```

Fixed with a one-liner:
```
[dev@d-lc9-kai build]$ sed -i 's|-Ios/windows/posix/include|-I$(SRCDIR)/os/windows/posix/include|g' .../Makefile 
[dev@d-lc9-kai build]$ make -j
....
  LINK t/fio-verify-state
  LINK t/stest
  LINK t/lfsr-test
  LINK fio
[dev@d-lc9-kai build]$ echo $?
0
```
I also verified that in-source builds continue to work with the modification.
```
[dev@d-lc9-kai fio]$ git status
HEAD detached at fio-3.39
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   Makefile

no changes added to commit (use "git add" and/or "git commit -a")
[dev@d-lc9-kai fio]$ make -j $(nproc) CROSS_COMPILE=x86_64-w64-mingw32-
```

As soon as I get to a stopping point in this project, I'll swing back around and open a PR, but I won't be upset if someone beats me to it.

--- a/Makefile
+++ b/Makefile
@@ -276,7 +276,7 @@ ifneq (,$(findstring CYGWIN,$(CONFIG_TARGET_OS)))
   SOURCE += os/windows/cpu-affinity.c os/windows/posix.c os/windows/dlls.c
   WINDOWS_OBJS = os/windows/cpu-affinity.o os/windows/posix.o os/windows/dlls.o lib/hweight.o
   LIBS	 += -lpthread -lpsapi -lws2_32 -lssp
-  FIO_CFLAGS += -DPSAPI_VERSION=1 -Ios/windows/posix/include -Wno-format
+  FIO_CFLAGS += -DPSAPI_VERSION=1 -I$(SRCDIR)/os/windows/posix/include -Wno-format
 endif
 
 ifdef cmdprio_SRCS

